packages/jsii-dotnet-generator/src/Amazon.JSII.Generator/VersionSuffixExtensions.cs

```diff
@@ -0,0 +1,26 @@
+using Amazon.JSII.JsonModel.Spec;
+
+namespace Amazon.JSII.Generator
+{
+    public static class VersionSuffixExtensions
+    {
+        public static string GetDecoratedVersion(this Assembly assembly)
+        {
+            return MakeDecoratedVersion(assembly.Version, assembly.Targets?.DotNet?.VersionSuffix);
+        }
+
+        public static string GetDecoratedVersion(this PackageVersion package)
+        {
+            return MakeDecoratedVersion(package.Version, package.Targets?.DotNet?.VersionSuffix);
+        }
+
+        private static string MakeDecoratedVersion(string version, string suffix)
+        {
+            if (suffix == null)
+            {
+                return version;
+            }
+            return $"{version}-{suffix}";
```

Comment:
Can we make this more consistent with java? Require that suffix will always begin with a `-` at the source